src/main/java/org/myrobotlab/service/InMoovHead.java

```diff
@@ -118,7 +119,7 @@ public boolean connect(String port) throws Exception {
   }
   
   public boolean connect(String port,Integer headYPin,Integer headXPin,Integer eyeXPin,Integer eyeYPin,Integer jawPin,Integer rollNeckPin) throws Exception {
-    arduino.connect(port);
+ // FIXME -  !!! =>  cannot do this "here" ??? arduino.connect(port);

```

Comment:
for backwards compatibility.. you should do a   if arduino instanceof Arduino  .. .. then cast it to an arduino and tell it to connect.

Reply:
ok ..  so there are many things which don't connect, and some things which connect but are not an Arduino .. 
So I think I might have an acceptable solution ..